pkg/controller/storagecluster/update.go

```diff
@@ -584,6 +584,8 @@ func matchSelectedFields(
 		return false, nil
 	} else if !reflect.DeepEqual(oldSpec.RuntimeOpts, currentSpec.RuntimeOpts) {
 		return false, nil
+	} else if !reflect.DeepEqual(oldSpec.Security, currentSpec.Security) {
```

Comment:
For security spec we will have to handle this a bit differently/selectively.
For instance, we should not restart the pods on guestMode change or token lifetime change. I think for the rest of the spec we will have to restart the pods.
May be just extract this out in a method just like `isEnvEqual`.

Reply:
Also add a test similar to `TestUpdateStorageCluster**` for security spec change.